### ISSUE TITLE
add hyperparameter optimization

### ISSUE THREAD

**Issue by naomi-simumba** [gridfm/gridfm-graphkit#26]
add functionality to perform hyperparameter optimization (HPO) and repeated seed runs during training/fine-tuning
[PR#23](https://github.com/gridfm/gridfm-graphkit/pull/23)
